src/stavenote.ts

```diff
@@ -223,11 +223,63 @@ export class StaveNote extends StemmableNote {
-            // upper voice is middle voice, so shift it right
-            noteU.note.setXShift(xShift + 3);
-          } else {
-            // shift lower voice right
+          //Vexflowpatch: Instead of shifting notes, remove the appropriate flag.
+          //If we are sharing a line, switch one notes stem direction.
+          //If we are sharing a line and in the same voice, only then offset one note
+          const lineDiff = Math.abs(noteU.line - noteL.line);
+          if (noteU.note.glyph.stem && noteL.note.glyph.stem) {
+            //If we have different dot values, must offset
+            //Or If we have a non-filled in mixed with a filled in notehead, must offset
+            if (
+              (noteU.note.duration === 'h' && noteL.note.duration !== 'h') ||
+              (noteU.note.duration !== 'h' && noteL.note.duration === 'h') ||
+              noteU.note.dots !== noteL.note.dots
+            ) {
+              noteL.note.setXShift(xShift);
+              if (noteU.note.dots > 0) {
+                let foundDots = 0;
+                for (const modifier of noteU.note.modifiers) {
+                  if (modifier instanceof Dot) {
+                    foundDots++;
+                    //offset dot(s) above the shifted note
+                    //lines + 1 to negative pixels
+                    modifier.setYShift(-10 * (noteL.maxLine - noteU.line + 1));
+                    if (foundDots === noteU.note.dots) {
+                      break;
+                    }
+                  }
+                }
+              }
+            } else if (lineDiff < 1 && lineDiff > 0) {
+              //if the notes are quite close but not on the same line, shift
+              noteL.note.setXShift(xShift);
+            } else if (noteU.note.voice !== noteL.note.voice) {
+              //If we are not in the same voice
+              if (noteU.stemDirection === noteL.stemDirection) {
+                if (noteU.line > noteL.line) {
+                  //noteU is above noteL
+                  if (noteU.stemDirection === 1) {
+                    noteL.note.renderFlag = false;
+                  } else {
+                    noteU.note.renderFlag = false;
+                  }
+                } else if (noteL.line > noteU.line) {
+                  //note L is above noteU
+                  if (noteL.stemDirection === 1) {
+                    noteU.note.renderFlag = false;
+                  } else {
+                    noteL.note.renderFlag = false;
+                  }
+                } else {
+                  //same line, swap stem direction for one note
+                  if (noteL.stemDirection === 1) {
+                    noteL.stemDirection = -1;
+                    noteL.note.setStemDirection(-1);
+                  }
+                }
+              }
+            } //Very close whole notes
+          } else if (!noteU.note.glyph.stem && !noteL.note.glyph.stem && lineDiff < 1.5) {
```

Comment:
@0xfe @sschmidTU I was trying to debug these lines in order to resolve the issue and I have realised that they do not get ever run. The function returns on line 151. There is a big FIXME there. It was already in 1.2.93.
Any ideas?

Reply:
Not sure what's happening here as I haven't debugged this in Vexflow recently and I don't know what's happening in line 151, but I'm pretty sure these lines were running withour patch in 1.2.93, because the whole patch did what it was supposed to do (see image above).

Reply:
the problem is now resolved in #1280